File: shared/types/lesson.ts
export type LessonStatus = 'scheduled' | 'completed' | 'cancelled' | 'rescheduled';

export interface Lesson {
  id: string;
  course_id: string;        // ссылка на курс-шаблон
  date: string;             // 2026-07-08
  time: string;             // 17:00 (legacy, может отличаться от курса)
  start_time: string;       // Новое: время начала (HH:MM)
  end_time: string;         // Новое: время окончания (HH:MM)
  title: string;            // может отличаться от курса
  status: LessonStatus;
  rescheduled_to: string;   // lesson_id, если перенесено
  homework: string;         // домашнее задание
  location: string;         // может отличаться от курса
  location_link: string;
  note: string;             // заметка преподавателя
  is_active: boolean;
  created_at: string;
}

File: shared/models/lesson.py
from enum import Enum
from pydantic import BaseModel
from typing import Optional


class LessonStatus(str, Enum):
    SCHEDULED = "scheduled"
    COMPLETED = "completed"
    CANCELLED = "cancelled"
    RESCHEDULED = "rescheduled"


class Lesson(BaseModel):
    id: str
    course_id: str = ""
    date: str
    time: str = ""               # Legacy: single time field, kept for backward compat
    start_time: str = ""         # New: lesson start time (HH:MM)
    end_time: str = ""           # New: lesson end time (HH:MM)
    title: str = ""
    status: LessonStatus = LessonStatus.SCHEDULED
    rescheduled_to: str = ""
    homework: str = ""
    location: str = ""
    location_link: str = ""
    note: str = ""
    is_active: bool = True
    created_at: str = ""
    user_id: str = ""
    updated_at: str = ""
